contracts/meta-oracles/LinearizedPriceDataSource.sol

```diff
@@ -84,27 +85,34 @@ contract LinearizedPriceDataSource is
      * exactly on nextAvailableUpdate do not compound as they would if it was required that poke is called
      * an updateInterval amount of time after the last poke.
      *
-     * @returns                Returns the datapoint from the Medianizer contract
+     * @param  _timeFromExpectedUpdate   Time passed from expected update
+     * @param  _updateInterval           Update interval of TimeSeriesFeed
+     * @param  _previousLoggedPrice      Previously logged price from TimeSeriesFeed
+     * @returns                          Returns the datapoint from the Medianizer contract
      */
-    function read()
+    function read(
+        uint256 _timeFromExpectedUpdate,
+        uint256 _updateInterval,
+        uint256 _previousLoggedPrice
```

Comment:
Sorry if my following thoughts are jumbled - but I see that we're trying to create a better separation and passing arguments is an improvement. However, I think passing arguments specific for the `LInearizedDataSource` use case is suboptimal - as it might not satisfy cases where these specific arguments aren't good enough (e.g. a datasource that needs N-2 logged price instead of previous).

We either pass all the state that any data source would ever need or none at all. Since we're leaning towards passing state (which I think makes sense), we can pass it all through a struct called something like `TimeSeriesState` and pass everything along (including timeSeriesData, updateInterval, etc.).

Reply:
I guess I'd argue we already are passing in all the relevant state with those three params. Perhaps we want to pass in an array of prices in case we want to use more than one in the future, but even that seems excessive. The only state "missing" is `nextEarliestUpdate` but that can be implied from `timeFromExpectedUpdate` and `updateInterval`. But I'd like all the math that uses `block.timestamp` to live on the TimeSeriesFeed if that makes sense.

Reply:
The variables being passed in right now are specific to the `LinearizedDataSource` requirements. I argue the job of `TimeSeriesFeed` is not to guess what the `DataSource` needs, but just provide any context it needs. 

The key question re :`block.timestamp` is: What is the job of the `TimeSeriesFeed` and what is the job of the `DataSource`?

For me, the job of `TimeSeriesFeed` is just to track data and get the data it needs. No logic at all for figuring out what the next data point is.

The job of `DataSource` is to get all the data it needs to calculate the data point and return that in the read function.